### PR TITLE
Return MessageReaction for getReactionX methods

### PR DESCRIPTION
**Pull request by duncte123**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: https://github.com/DV8FromTheWorld/JDA/projects/5#card-76582150

## Description

Makes `Message#getReactionX` return `MessageReaction` instead of `MessageReaction.ReactionEmote`

### CODE REVIEW

**Comment by java-coding-prodigy** [comment]
#2015 seems to cover this issue.

**Comment by duncte123** [comment]
> #2015 seems to cover this issue.

That one only touches the mentions, this is specifically for emotes

**Comment by java-coding-prodigy** [comment]
> That one only touches the mentions, this is specifically for emotes

Oh I see.

File: src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java [hunk 1]
```diff
@@ -304,29 +304,27 @@ public ReactionPaginationAction retrieveReactionUsers(@Nonnull String unicode)
...
         return this.reactions.stream()
-            .map(MessageReaction::getReactionEmote)
-            .filter(r -> r.isEmoji() && r.getEmoji().equals(unicode))
+            .filter(r -> r.getReactionEmote().isEmoji() && r.getReactionEmote().getEmoji().equals(unicode))
             .findFirst().orElse(null);
     }
 
     @Override
-    public MessageReaction.ReactionEmote getReactionById(@Nonnull String id)
+    public MessageReaction getReactionById(@Nonnull String id)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java:318]: I saw you already mentioned this on discord, can you make this a default in the interface?

**Review by MinnDevelopment — APPROVED** [review]

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 2]
```diff
@@ -2358,13 +2371,17 @@ default MessageAction reply(@Nonnull byte[] data, @Nonnull String name, @Nonnull
      * @throws java.lang.UnsupportedOperationException
      *         If this is a system message
      *
-     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} of this message or null if not present.
+     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction MessageReaction} of this message or null if not present.
+     *
+     * @see #getReactionById(String)
+     * @see #getReactionByUnicode(String)
+     * @see net.dv8tion.jda.api.entities.MessageReaction
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:2378]:
```suggestion
     * @see    #getReactionById(String)
     * @see    #getReactionByUnicode(String)
     * @see    net.dv8tion.jda.api.entities.MessageReaction
```

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 3]
```diff
@@ -2336,20 +2342,27 @@ default MessageAction reply(@Nonnull byte[] data, @Nonnull String name, @Nonnull
      * @throws java.lang.IllegalArgumentException
      *         If the provided id is not a valid snowflake.
      *
-     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} of this message or null if not present.
+     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction MessageReaction} of this message or null if not present.
+     *
+     * @see #getReactionById(long)
+     * @see #getReactionByUnicode(String)
+     * @see net.dv8tion.jda.api.entities.MessageReaction
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:2349]:
```suggestion
     * @see    #getReactionById(long)
     * @see    #getReactionByUnicode(String)
     * @see    net.dv8tion.jda.api.entities.MessageReaction
```

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 4]
```diff
@@ -2312,20 +2315,23 @@ default MessageAction reply(@Nonnull byte[] data, @Nonnull String name, @Nonnull
      * @throws java.lang.IllegalArgumentException
      *         If the provided unicode value is null or empty.
      *
-     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction.ReactionEmote ReactionEmote} of this message or null if not present.
+     * @return The {@link net.dv8tion.jda.api.entities.MessageReaction MessageReaction} of this message or null if not present.
+     *
+     * @see #getReactionById(long)
+     * @see net.dv8tion.jda.api.entities.MessageReaction
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:2321]:
```suggestion
     * @see    #getReactionById(long)
     * @see    net.dv8tion.jda.api.entities.MessageReaction
```

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 5]
```diff
@@ -963,6 +964,8 @@ default Button getButtonById(@Nonnull String id)
      *         If this is a system message
      *
      * @return Immutable list of all MessageReactions on this message.
+     *
+     * @see net.dv8tion.jda.api.entities.MessageReaction
```
Comment (Tais993) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:968]:
```suggestion
     * @see    net.dv8tion.jda.api.entities.MessageReaction
```

**Review by Tais993 — CHANGES_REQUESTED** [review]
According to the [structure-guide ](https://github.com/DV8FromTheWorld/JDA/wiki/6%29-JDA-Structure-Guide) they should be aligned.

_I do acknowledge that this doesn't happen everywhere though, there's a lot of inconsistencies with the indentation of javadoc_

**Review by Sanduhr32 — APPROVED** [review]

**Review by DV8FromTheWorld — APPROVED** [review]

**Pull request merged**